### ISSUE TITLE
Slow creation of static DB roles

### ISSUE THREAD

**Issue by jacobkiers** [hashicorp/vault#7467]
**Describe the bug**

We use a script that configures Vault during bootup. One of the things configured, are static DB roles. With many (but not all) of these, there is a slowdown, after which the request is cancelled.

**To Reproduce**

1. `vault write app/db/config/app allowed_roles="s-my-role-1 s-my-role-2"`

The following  command runs in a loop for the different roles: 

1. Run `vault write app/db/static-roles/s-my-role-2 db_name="app" rotation_statements="GRANT ALL ON *.* TO '{{name}}'@'%' IDENTIFIED BY '{{password}}';" username="my-role" rotation_period=24h`

And with some (not all), the command errors with the following message:

`Error writing data to dashboard/db/static-roles/s-my-role: Put https://172.18.0.7:8200/v1/app/db/static-roles/s-my-role: net/http: request canceled (Client.Timeout exceeded while awaiting headers)`

**Expected behavior**

The role is created (quickly).

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.2.2
* Vault CLI Version (retrieve with `vault version`): Vault v1.2.2
* Server Operating System/Architecture: Docker (v1.10.1) with image vault:1.2.2 on Debian Stretch
* MySQL version: 5.6

Vault server configuration file(s):

```hcl
storage "mysql" {
  address = "172.31.255.254:3306"
  username = "root"
  password = "dev-root"
  database = "vault"
}

listener "tcp" {
  address = "0.0.0.0:8200"
  tls_cert_file = "/vault/config/dev.chain.pem"
  tls_key_file = "/vault/config/dev.key"
}

api_addr = "https://app.cmd:8200"
disable_mlock = "true"
```

**Comment by ncabatoff** [comment]
Hi @jacobkiers,

Thanks for the bug report.

Fixed in #7518.

**Comment by jacobkiers** [comment]
Hi @ncabatoff, thanks for looking into it. Is there a nightly or something with which I can verify that it is actually solved? 

The linked MR seems to address a slightly different issue, updating the _same_ role multiple times. This issue is about _many different_ roles being created in a batch.

I'm not well versed enough in go to evaluate whether the code addresses both.
